enhancements/sig-architecture/47-manifestwork-updatestrategy/README.md

```diff
@@ -0,0 +1,223 @@
+3. user can choose to use server side apply when work-agent update the resource.
+4. to support lowner version of kube, we may also cosider allowing user to choose json merge approach to update the resource.
+
+### Design Details
+
+#### API change
+
+We could add an updateStraty for each manifest as below:
+
+```go
+type UpdateStrategy struct {
+	// type defines the strategy to update this manifest, default value is Update.
+	// Update type means to update resource by an update call.
+	// None type means do not update resource based on current manifest.
+	// ServerSideApply type means to update resource using server side apply as fieldManager of work-controller.
+	// If there is conflict, the related Applied confition of manifest will in the status of False with the
+	// reason of ApplyConflict.
+	// +kubebuilder:default=60
+	// +kubebuilder:validation:Enum=Update;None;ServerSideApply
+	// +kubebuilder:validation:Required
+	// +required
+	Type string `json:"type"`
+
+	// serverSideApply defines the configuration for server side apply. It is honored only when
+	// type of updateStrategy is ServerSideApply
+	// +optional
+	ServerSideApply *ServerSideApplyConfig `json:"serverSideApply,omitempty"`
+}
+
+type ServerSideApplyConfig struct {
+	// Force represents to force apply the manifest.
+	// +optional
+	Force bool `json:"force"`
+}
+```
+
+And example of the API would be:
+
+```yaml
+kind: ManifestWork
+metadata:
+  name: demo-work1
+spec:
+  workload:
+    manifests:
+    - apiVersion: apps/v1
+      kind: Deployment
+      metadata:
+        name: hello
+        namespace: default
+      spec:
+        replica: 1
+        selector:
+          matchLabels:
+            app: hello
+        template:
+          metadata:
+            labels:
+              app: hello
+          spec:
+            containers:
+            - name: hello
+              image: quay.io/asmacdo/busybox
+              command: ['sh', '-c', 'echo "Hello, World!" && sleep 3600']
+  manifestConfigs:
+  - resourceIdentifier:
+      group: apps
+      resource: deployments
+      name: hello
+      namespace: default
+    updateStrategy:
+      type: ServerSideApply   
+```
+
+It means that the to create deployment `hello` on the managed cluster if it does not exist, and using server side apply to update
+the resource if it exists on the managed cluster already.
+
+#### Server side apply
+
+With service side apply, it is possible to make work agent coordinate with other actors on the spoke. Take the above `ManifestWork` as an
+example, if there is an HPA controller on the spoke that take the ownership of the `spec.replicas` field. The condition of `ManifestWork`
+should show a status condition of applied failed as below:
+
+```yaml
+status:
+  resourceStatus:
+    manifests:
+      - conditions:
+        - lastTransitionTime: "2021-10-14T14:59:09Z"
+          message: Manifest ownership conflict
```

Comment:
we can probably reflect the name of the conflicting field-manager in the status message for clarity?

Reply:
additionally, we can provide a better message to those users whose managed cluster doesnt support SSA. this condition can be detected by parsing the returned error after sending SSA requests.